app/jobs/google/search_keyword_job.rb

```diff
@@ -15,6 +15,8 @@ def update_keyword(keyword, search_result)
         keyword.update search_result.merge(status: :parsed)
       else
         keyword.update({ status: :failed })
+
+        raise Google::Errors::SearchKeywordError unless search_result
```

Comment:
```suggestion
        raise Google::Errors::SearchKeywordError
```

This condition seems duplicated with the `it/else` (as is is already in the `else` block that ensures `search_result` is falsy. 

But why exactly do we want to raise this error? 👀 
I do not see any reason why we need to raise an error here: as we already handle the case by setting the keyword status to `failed`, it seems to me that we do not need to consume more resources with raising an error. 🤔

Reply:
@malparty Sidekiq comes with a [built-in retry](https://www.rubydoc.info/gems/sidekiq/Sidekiq/JobRetry) feature. By raising an error in the job, Sidekiq will create a retry queue for that job. We set status fail so that users and backend know it is already fetched but fail. 🙏

Reply:
Oh yeah makes sense.

Don't forget to remove the useless `unless`  condition though! :D 

```suggestion
        raise Google::Errors::SearchKeywordError
```

Reply:
Nice catch.